### ISSUE TITLE
Long/complex pages cause formatting errors

### ISSUE THREAD

**Issue by cpmiller51** [turnermm/ckgedit#215]
Have seen this on a couple of occasions now. A long/complex page created in markup is formatted correctly. Convert the page to CKGEDIT and save it--make no changes, just save it--and formatting gets messed up. For example, spaces between words are removed, boldface is removed, and similar. This has been a consistent problem over the past four or five months. (And I'm only reporting it now because it didn't much concern me before as I was learning the markup language.)

Two files are attached.

The first is a text file containing the properly formatted markup as originally created. No problems here.

The second is a PDF. The first five pages are the displayed content correctly formatted as they come out of the Dokuwiki markup source (see the text file). The next five pages (that start with a pop-up note) are the same pages after being mauled by CKGEDIT. The mistakes are all highlighted in green and occur in the last few pages.

The source wiki is closed to the public but if you'd like access, please ask and I'll make arrangements. Any other questions, ask away.

Charles Miller
[DOKUWIKI SOURCE FOR CKGEDIT.txt](https://github.com/turnermm/ckgedit/files/331211/DOKUWIKI.SOURCE.FOR.CKGEDIT.txt)
[DOKUWIKI Correct & Incorrect Formatting.pdf](https://github.com/turnermm/ckgedit/files/331213/DOKUWIKI.Correct.Incorrect.Formatting.pdf)

**Comment by turnermm** [comment]
First, I can't determine anything about the second 5 pages just by looking at it in the pdf.  I'd need the dokuwiki text file.  But I suspect that it may not be the length but something in the second section that is incompatible with ckgedit.  Probably the highlighted text.  What plugin are you using for that?  And are you using any other plugins there that you are not using in the first 5 pages?

Secondly, because of the wide variety of possibilities for incompatibilities, there is no guarantee that every dw source file will convert correctly into ckgedit.

**Comment by turnermm** [comment]
I just re-read your comment and see I  misunderstood -- it hadn't been clear to me that your pdf was showing two versions of the same page.  I'll put the text file into one of my test installs and see what heppens.

**Comment by turnermm** [comment]
The "all messed up" is the removal of bolding?  Or is there something else?

**Comment by cpmiller51** [comment]
Both the removal of the bolding AND the removal of blanks between words. If
you look at the PDF with the error markup, you'll see both types of errors
under the green highlighting. I didn't do a line-by-line comparison, but a
quick skim seemed to show the same errors occurring after I removed the
NOTE plugin as before.

**Comment by turnermm** [comment]
Note isn't the problem

Get Outlook for iOS




On Fri, Jun 24, 2016 at 10:13 AM -0500, "cpmiller51" <notifications@github.com> wrote:










Both the removal of the bolding AND the removal of blanks between words. If

you look at the PDF with the error markup, you'll see both types of errors

under the green highlighting. I didn't do a line-by-line comparison, but a

quick skim seemed to show the same errors occurring after I removed the

NOTE plugin as before.



On Fri, Jun 24, 2016 at 11:10 AM, Myron Turner <notifications@github.com>

wrote:

**Comment by cpmiller51** [comment]
Huh? "Get Outlook for iOS"???

Am I missing something important here?

**Comment by turnermm** [comment]
Just my phone making life difficult.  

Was trying to say that Note is not the problem.

**Comment by turnermm** [comment]
Try the [captcha branch](https://github.com/turnermm/ckgedit/archive/captcha.zipl). Please report back. 

One reason I misread your original is that I was looking for "all messed up", not the loss of  a dw format under limited circumstances.

**Comment by cpmiller51** [comment]
Your link is a 404. Aside from that, I'm not the brightest bulb in the box when it comes to installing this stuff. I'm not a developer or a Linux guy by training or inclination--I was data centers, telecomms, and IT security.

Do I just feed the URL that you'll provide into the manual install option under the DK extension manager and let 'r rip?

**Comment by turnermm** [comment]
Corrected Link: https://github.com/turnermm/ckgedit/archive/captcha.zip.  Yes. just paste this link into the manual install tab and then click the ``install`` tab.  This branch enables captchas using the captcha plugin; otherwise it's fully compatible with the master.   The captcha will not work unless you have the captcha plugin installed and it will not affect logged in users, unless the captcha plugin's ``forusers``  option is set to true.

**Comment by cpmiller51** [comment]
Myron

Did the install of the catcha version. It seems to work, at least with the
test page we've been looking at. I'll work with this for a few days on some
other longer/more complex pages and see how things go. Look for a follow-on
report by, oh, Tuesday coming? Sound like a plan?

Thanks, Myron. Appreciate the quick response.

Charles.

**Comment by turnermm** [comment]
It's not the length, but some uncaught bug that may show up in a longer document but would be there no matter how long or short the document.

Looking forward to your report.

**Comment by cpmiller51** [comment]
Hi Myron

Sorry, I'm a couple days late and a few dollars short on getting back to
you by Tuesday.

I'm diddled and fiddled with tables and formatting inside a couple of my
more complex articles, and editing all manner of other bits and pieces
using the captcha version. To date, I haven't found anything that doesn't
work (one exception, which I'll get to in a minute) or any other formatting
anomalies like those I was finding with the other version, already
discussed.

The one exception. I think. I only think because who knows...it could be me
misunderstanding something.

Internal links.

The intent here is to create an internal (to the current page) link,
linking selected text to an already-defined subheading on the same page.

   1. Highlight some text.
   2. Click the link button.
      - The default link type is "URL."
   3. Change the link type to "Internal link."
   4. Noting the "Use the advanced tab to create page anchors and query
   strings" note,
   5. click the Advanced tab.
   6. Click the Get Headings button per the instructions at the top of the
   dialog.
   7. Nothing happens.

So, that'd be the only issue I discovered.

Note: I did the testing under both the old Detritus and the new "Elenor of
Tsort" (THAT one just rolls of the tongue now, doesn't it?) releases. I did
NOT update the CKGEDIT release. IOW, I had the captcha version, was offered
the 6/28 "master" version, and didn't take it. Please advise on whether I
should change to the latest master version.

Thanks, Myron!

Charles

**Comment by turnermm** [comment]
Switch to master. The link is a separate issue

**Comment by turnermm** [comment]
Just tested on IE.  No problems.  The query string and the anchor text do not show up in the editor; they are embedded in the link url.  Maybe that's your mistake.

**Comment by cpmiller51** [comment]
Hi Myron

I switched to the "master" version and ran some tests. I couldn't break it
so, for now at least, I think whatever the bug was, you've fixed it.

WRT the create link issue, the fault lay with me. It wasn't clear to me
that I had to FIRST pick a page from the "Browse server" dialog and THEN go
to the advanced tab. I simply assumed the current page was going to get
used by default. Once I got past my d'oh moment, it all seems to work.

Thanks

Charles
